plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetPersister.java

```diff
@@ -627,6 +627,7 @@ public Throwable getError() {
         @Override
         protected IStatus run(DBRProgressMonitor monitor) {
             model.setUpdateInProgress(this);
+            UIUtils.syncExec(viewer::fireResultSetChange); // Update "save" and "cancel" buttons
```

Comment:
Can we fire this event outside the `DataUpdaterJob` job, so we can avoid using `syncExec` here?